flocker/provision/test/test_install.py

```diff
@@ -411,3 +415,69 @@ def test_with_branch_and_version(self):
             run(command="yum install --enablerepo=clusterhq-build "
                         "-y clusterhq-flocker-node-1.2.3-1")
         ]))
+
+
+class PrivateKeyLoggingTest(SynchronousTestCase):
+
+    def test_private_key_removed(self):
+        """
+        A private key is removed for logging.
+        """
+        key = dedent('''
+            -----BEGIN PRIVATE KEY-----
+            MFDkDKSLDDSf
+            MFSENSITIVED
+            MDKODSFJOEWe
+            -----END PRIVATE KEY-----
+            ''')
+        self.assertIn('SENSITIVE', key)
+        logged = _remove_private_key(key)
+        self.assertIn('-----BEGIN PRIVATE KEY-----', logged)
+        self.assertNotIn('SENSITIVE', logged)
+        self.assertIn('-----END PRIVATE KEY-----', logged)
+
```

Comment:
Similar to other comments, this can just be one assert on the result of the transformed key for logging. Even `dedent` isn't strictly necessary, in that

```
        key = ("\n-----BEGIN PRIVATE KEY-----"
               "\nMFDkDKSLDDSf"
               "\nMFSENSITIVED"
               "\nMDKODSFJOEWe"
               "\n-----END PRIVATE KEY-----\n")
```

produces the same result, but even with `dedent` I don't think we need to test that function's behaviour by asserting it hasn't removed "SENSITIVE" from a hardcoded string.